### PR TITLE
Added 8-bit FRB injection code

### PR DESCRIPTION
**Pull request by RaghavWani**

### CODE REVIEW

**Comment by RaghavWani** [comment]
Adding 8bit injection

File: arachne.c [hunk 1]
```diff
@@ -493,51 +493,1343 @@ int main(int argc, char *argv[]) {
...
-              out = 1;
-            else
-              out = 0;
-          }
-          raw[I] = out;
+         /*========================================================================*/
+         /*======================== 8 bit FRB Injection ===========================*/
+          switch (in) {
+          case 0:
+            for (m = 255-in; m > 0; m--) {
```
Comment (astrogewgaw) [inline arachne.c:500]: You need to declare `m` as an integer here, like so:

```c
for (int m = 255-in; m > 0; m--) {
```

File: arachne.c [hunk 2]
```diff
@@ -493,51 +493,1343 @@ int main(int argc, char *argv[]) {
...
+              plvl = (prob(min((in -127)*lvl, (in + m -127)*lvl - signal)) - prob(max((in -128)*lvl, (in + m -128)*lvl - signal)))/
```
Comment (astrogewgaw) [inline arachne.c:501]: You have not declared `plvl` anywhere, you need to modify this line:

https://github.com/astrogewgaw/arachne/blob/59b44411d6fe6bca8603a0f67542b034e9e6bc6e/arachne.c#L492

to:

```c
double plvl;
```

**Review by astrogewgaw — CHANGES_REQUESTED** [review]
Feels like doing this as a switch case was a mistake. I think what we should do instead it isolate this into a function that takes the input variable `in` and the probability `pval` as inputs, and just returns the modified value, which we then equate to `out`.

File: arachne.c (hunk 2 above)
Comment (RaghavWani) [reply arachne.c:501]: Oh okay, I will make these changes.

**Pull request closed without merging**